tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -297,21 +297,26 @@ static std::string RetTypeOfLambda(const std::string &lambdaName)
    return type;
 }
 
-
 static void GetTopLevelBranchNamesImpl(TTree &t, std::set<std::string> &bNamesReg, ColumnNames_t &bNames,
-                                       std::set<TTree *> &analysedTrees)
+                                       std::set<TTree *> &analysedTrees, const std::string friendName = "")
 {
-
    if (!analysedTrees.insert(&t).second) {
       return;
    }
 
    auto branches = t.GetListOfBranches();
    if (branches) {
       for (auto branchObj : *branches) {
-         auto name = branchObj->GetName();
+         const auto name = branchObj->GetName();
          if (bNamesReg.insert(name).second) {
             bNames.emplace_back(name);
+         } else if (!friendName.empty()) {
```

Comment:
Is the `} else` warranted?  Or should the code always (try to) insert (also) the name with the alias?

Reply:
I think it's warranted, we don't want duplicated entries in the list of branches. If possible, we add the branch with its name -- otherwise (if a branch with that name already exists in the main tree) we add it as `friendname.branchname`.

Before this patch, we were only adding the friend branches that had names different from those of branches in the main tree.

Reply:
Yes but ... if the branch is in a friend and has a unique name you will be adding only "branchname" .. what happen if the user uses "friendname.branchname".

Reply:
ah I understand the misunderstanding now: GetTopLevelBranchNames is not used by RDataFrame to create the list of valid column names, it's only used by Snapshot and Cache to create the default list of columns to write out to disk (overridable by passing a list of column names explicitly).